docs/flow/components/local.file.md

```diff
@@ -0,0 +1,91 @@
+# local.file
+
+The `local.file` component exposes the contents of a file on disk to other
+components. The file will be watched for changes so that its latest content is
+always exposed.
+
+The most common use of `local.file` is to load secrets (e.g., API keys) from
+files.
+
+Multiple `local.file` components can be specified by giving them different name
+labels.
+
+## Example
+
+```hcl
+local "file" "my-file" {
+  filename = "path/to/my/file"
+}
+```
+
+## Arguments
+
+The following arguments are supported and can be referenced by other
+components:
+
+Name | Type | Description | Default | Required
+---- | ---- | ----------- | ------- | --------
+`filename` | `string` | Path of the file on disk to watch | | **yes**
+`detector` | `string` | Which file change detector to use (fsnotify, poll) | `"fsnotify"` | no
+`poll_frequency` | `duration` | How often to poll for file changes | `"1m"` | no
+`sensitive` | `bool` | Marks the file as [sensitive][] | `false` | no
+
+### File change detectors
+
+File change detectors are used for detecting when the file needs to be re-read
+from disk. `local.file` supports two detectors: `fsnotify` and `poll`.
+
+#### fsnotify
+
+The `fsnotify` detector subscribes to filesystem events which indicate when the
+watched file had been updated. This requires a filesystem which supports events
+at the Operating System level: network-based filesystems like NFS or FUSE won't
+work.
+
+When a filesystem event is received, the component will reread the watched
+file. This will happen for any filesystem event to the file, including a change
+of permissions.
+
+`fsnotify` also polls for changes to the file with the configured
+`poll_frequency` as a fallback.
+
+`fsnotify` will stop receiving filesystem events if the watched file has been
+deleted, renamed, or moved. The subscription will be re-established on the next
+poll once the watched file exists again.
+
+#### poll
+
+The `poll` file change detector will cause the watched file to be reread
+every `poll_frequency`, regardless of whether the file changed.
+
+## Exported fields
+
+The following fields are exported and can be referenced by other components:
+
+Name | Type | Description
+---- | ---- | -----------
+`content` | `string` or `secret` | The contents of the file from the most recent read
```

Comment:
should we use the term sensitive here too?

Reply:
The type is called `secret` internally, maybe we should use `secret` consistently instead of sensitive?

Reply:
Sounds good